tutorials/eda_contents/Computing_Environment/Unix.md

```diff
@@ -0,0 +1,363 @@
+
+ 1. Don't use whitespaces.
+
+   Whitespaces can make a name more meaningful but since whitespace is used to break arguments on the command line
+   is better to avoid them on name of files and directories. You can use `-` or `_` instead of whitespace.
+
+ 2. Don't begin the name with `-` (dash).
+
+    Commands treat names starting with `-` as options.
+
+ 3. Stick with letters, numbers, `.` (period), `-` (dash) and `_` (underscore).
+
+    Many other characters have special meanings on the command line. We will learn
+    about some of these during this lesson. There are special characters that can
+    cause your command to not work as expected and can even result in data loss.
+
+ If you need to refer to names of files or directories that have whitespace
+ or another non-alphanumeric character, you should surround the name in quotes (`""`).
+
+ Now, let's go to the new directory we have created.
+
+Assuming you made a directory called `example_of_directory`, then type:
+
+~~~
+$ cd ex<TAB>
+~~~
+
+This will auto-input `example_of_directory`.
+
+~~~
+$ pwd
+/notebook_dir/mypublic/example_of_directory
+~~~
+Let's create a file called `new_file.txt` in this new dictionary using the command `touch`.
+
+~~~
+/example_of_directory$ touch new_file.txt
+/example_of_directory$ ls
+new_file.txt
+~~~
+
+Now, let's see how we can remove any files using the `rm` command:
+
+~~~
+$ rm new_file.txt
+$ ls
+$ 
+~~~
+
+The `ls` command now returns no output. 
+
+#### Deleting Is Forever
+
+The Unix shell doesn't have a trash bin that we can recover deleted files from (though 
+most graphical interfaces to Unix do).  Instead, when we delete files, they are 
+unhooked from the file system so that their storage space on disk can be recycled.
+Tools for finding and recovering deleted files do exist, but there's no guarantee
+they'll work in any particular situation, since the computer may recycle the 
+file's disk space right away.
+
+Note, `rm` can only remove files with only remove files in the current working 
+directory. `rm`, alone, cannot remove directories. To do this, we can use 
+the option `-r` or `-R` or `--recursive` which recursively deletes all directories 
+and their contents.
+
+### With Great Power Comes Great Responsibility
+
+ Removing the files in a directory recursively can be very dangerous
+ operation. If we're concerned about what we might be deleting we can
+ add the "interactive" flag `-i` to `rm` which will ask us for confirmation
+ before each step
+
+~~~
+ $ rm -r -i new_file.txt
+ rm: remove regular empty file 'new_file.txt'? y
+~~~
+
+### Some Unix Commands
+
+|**User Action**|**Command**|
+|List| ls *option*|
+|Change Directory| cd *option*|
+|Print Working Directory| pwd|
+|Username| whoami|
+|Host name| hostname|
+|Make Directory| mkdir *name*|
+|Make file| touch *name*|
+|Delete| rm *option* *file_path*|
+|Move file| mv *old_path* *new_path*|
+
+To learn more about the functions of a command, use `command --help` 
```

Comment:
Not all commands have a `--help` option. I'd recommend using `man` over `--help` (or recommend both)